File: accsaber-database/src/main/java/de/ixsen/accsaber/database/views/AccSaberPlayer.java
package de.ixsen.accsaber.database.views;

import de.ixsen.accsaber.database.views.extenders.WithRankAndStats;
import org.hibernate.annotations.Immutable;

import javax.persistence.Id;
import javax.persistence.MappedSuperclass;

@MappedSuperclass
@Immutable
public abstract class AccSaberPlayer extends WithRankAndStats {

    @Id
    private String playerId;
    private String playerName;
    private String avatarUrl;
    private Boolean isAccChamp;
    private String hmd;

    public String getPlayerId() {
        return this.playerId;
    }

    public String getPlayerName() {
        return this.playerName;
    }

    public String getAvatarUrl() {
        return this.avatarUrl;
    }

    public Boolean getAccChamp() {
        return this.isAccChamp;
    }

    public String getHmd() {
        return this.hmd;
    }
}

File: accsaber-database/src/main/kotlin/de/ixsen/accsaber/database/views/AccSaberPlayer.kt
package de.ixsen.accsaber.database.views

import de.ixsen.accsaber.database.views.extenders.WithRankAndStats
import org.hibernate.annotations.Immutable
import javax.persistence.Id
import javax.persistence.MappedSuperclass

@MappedSuperclass
@Immutable
abstract class AccSaberPlayer : WithRankAndStats() {
    @Id
    val playerId: Long? = null
    val playerName: String? = null
    val avatarUrl: String? = null
    val isAccChamp: Boolean? = null
    val hmd: String? = null
    val rankingLastWeek: Long? = null
}